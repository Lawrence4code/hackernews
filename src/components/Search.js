import React from 'react';

// Search Input Component
const Search = ({ value, onChange, onSubmit, children }) => (
  <form className="header__search" onSubmit={onSubmit}>
    <label htmlFor="search"> Search</label>
    <input
      id="search"
      type="text"
      value={value}
      onChange={onChange}
      placeholder="Search input..."
    />
    <button type="submit">{children}</button>
  </form>
);

export default Search;
