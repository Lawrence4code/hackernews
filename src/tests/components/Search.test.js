import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../../components/Search';

test('should render Search component correctly', () => {
  const component = renderer.create(<Search />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
