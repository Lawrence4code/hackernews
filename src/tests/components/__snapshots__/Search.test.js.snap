// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render Search component correctly 1`] = `
<form
  className="header__search"
>
  <label
    htmlFor="search"
  >
     Search
  </label>
  <input
    id="search"
    placeholder="Search input..."
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;
